### PR TITLE
Fix soundness issue with bags.card when a child appears multiple times in bag.union_disjoint

### LINKED ISSUE #9780: Segmentation fault on `Bags`
Hi,
For this instance, cvc5 https://github.com/cvc5/cvc5/commit/f8c6f02c1bedb8973254505affe3f21f50c2e8fc
```
$ cvc5 small.smt2 
cvc5 suffered a segfault in DEBUG mode.
Offending address is 0x0
Looks like a NULL pointer was dereferenced.
Segmentation fault (core dumped)
$ cat small.smt2 
(set-logic ALL)
(declare-const x Int)
(declare-fun v () (Bag Int))
(declare-fun a () String)
(assert (or (= 0 (bag.card v)) (= (str.substr a 1 0) (str.substr (str.from_int 0) (* x x) 1))))
(check-sat)
```

### PR DESCRIPTION
**Pull request by mudathirmahgoub**
Fixes #9780.
Fixes https://github.com/cvc5/cvc5-projects/issues/547.
Fixes https://github.com/cvc5/cvc5-projects/issues/566.
Fixes https://github.com/cvc5/cvc5-projects/issues/584.
Fixes https://github.com/cvc5/cvc5-projects/issues/603.

### CODE REVIEW

**Review by ajreynol — CHANGES_REQUESTED** [review]
You'll need to add the 2 new regressions to `test/regress/cli/CMakeLists.txt`.
Also can you add the test from https://github.com/cvc5/cvc5-projects/issues/656 (to https://github.com/cvc5/cvc5/tree/main/test/api/cpp).

File: test/regress/cli/regress1/bags/proj-issue584.smt2 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+(set-logic ALL)
+(set-info :status sat)
+(set-option :sygus-rr-synth-input true)
```
Comment (ajreynol) [inline test/regress/cli/regress1/bags/proj-issue584.smt2:3]: This option does not exist anymore, can you remove it?

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: test/regress/cli/regress1/bags/proj-issue584.smt2 (hunk 1 above)
Comment (mudathirmahgoub) [reply test/regress/cli/regress1/bags/proj-issue584.smt2]: Done

File: test/api/cpp/proj-issue656.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include <cvc5/cvc5.h>
```
Comment (ajreynol) [inline test/api/cpp/proj-issue656.cpp:1]: I believe this should have a cvc5 header like the other unit tests in this directory.

**Review by ajreynol — CHANGES_REQUESTED** [review]
One last minor comment.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**